Makefile

```diff
@@ -218,19 +218,20 @@ ifdef SSH_AUTH_SOCK
   EXTRA_DOCKER_ARGS += -v $(SSH_AUTH_SOCK):/ssh-agent --env SSH_AUTH_SOCK=/ssh-agent
 endif
 
-# Volume-mount gopath into the build container if it's explicitly set for persistent caching.
+# Volume-mount gopath into the build container to cache go module's packages. If the environment is using multiple
+# comma-separated directories for gopath, use the first one, as that is the default one used by go modules.
 ifneq ($(GOPATH),)
-# CircleCI's gopath is readonly and readonly gopaths are incompatible with go modules so don't
-# volume mount the cache in that environment
-ifndef CIRCLECI
 	# If the environment is using multiple comma-separated directories for gopath, use the first one, as that
 	# is the default one used by go modules.
-	LOCAL_GOPATH = $(shell echo $(GOPATH) | cut -d':' -f1)
-	EXTRA_DOCKER_ARGS += -v $(LOCAL_GOPATH)/pkg/mod:/go/pkg/mod:rw
-endif
+	GOMOD_CACHE = $(shell echo $(GOPATH) | cut -d':' -f1)/pkg/mod
+else
+	# If gopath is empty, default to $(HOME)/go.
+	GOMOD_CACHE = $(HOME)/go/pkg/mod
```

Comment:
But if the user doesn't have GOPATH, I guess that that path won't exist?

Reply:
I feels like it would be more right only to extend EXTRA_DOCKER_ARGS when GOPATH exists.

Reply:
There's some code in this PR to create that directory, so this should be fine.

Reply:
Sorry, I see that now, so great.